### ISSUE TITLE
File is not redownloaded even though URLs are different and file sizes are slightly different!

### ISSUE THREAD

**Issue by sidgo28** [lingochamp/FileDownloader#428]
I have downloaded a file named "Translations.zip" with a full path "fullParentPath/Translations.zip". The file is 20,000 bytes. 

Now, when I download new version of the file with the same name (and the same full path), but with different URL and slightly different file size (20,100 bytes), the downloader says that the download is complete even though nothing is downloaded (that is, the old file is not replaced with the new file).

In short, the downloader does not even attempt to download the new file, but wrongly identifies the old file as the new file.

**Comment by Ernest-su** [comment]
FileDownloader doesn't store download info when download file complete. so I guess you should check   file yourself or force FileDownloader redownload it.

**Comment by MoQ93** [comment]
@Ernest-su How can I force FileDownloader to download a file? Thanks!

**Comment by Jacksgong** [comment]
If the file you provided exists FileDownloader will callback `completed` immediately since v0.3.3, that it is. Because if file is on download processing, there is a temp suffix after the file name, so we can pick it out. Ref: https://github.com/lingochamp/FileDownloader/issues/172#issuecomment-223903250

---

#### Solution

If you want to download file ignore whether it the file has already existed, just feel free to invoke `BaseDownloadTask#setForceReDownload(true)`.

```java
/**
 * Force re-downloading the file regardless the target file is exist.
 *
 * @param isForceReDownload If set to true, will not check whether the target file is exist.
 *                          <p>
 *                          Default: {@code false}.
 */
BaseDownloadTask setForceReDownload(final boolean isForceReDownload);
```
